File: Graph/Floyd_warshall.py
# Find shortest distances between every pair of vertices in a given edge weighted directed Graph
def floyd_warshall(graph):
    distance_matrix = []
    # for storing resultant shortest distances
    for i in graph:  # type: float
        distance_matrix.append(i)
    # Number of vertices in graph

    V = len(graph) - 1
    # k is an intermediate vertex
    for k in range(V + 1):
        # i is the source
        for i in range(V + 1):
            # j is the destination
            for j in range(V + 1):
                # store the path which is shorter: min(i->j, i->k->j)
                if distance_matrix[i][k] == math.inf or distance_matrix[k][j] == math.inf:
                    pass
                else :
                    distance_matrix[i][j] = min((distance_matrix[i][j]), float(distance_matrix[i][k]) + float(distance_matrix[k][j]))
    # return the resultant matrix
    return distance_matrix


import math

if __name__ == '__main__':
    infinity = math.inf
    first_row = list(map(lambda x: infinity if x == 'inf' else float, input("enter the first row").split(" ")))
    second_row = list(map(lambda x: infinity if x == 'inf' else float, input("enter second row ").split(" ")))
    third_row = list(map(lambda x: infinity if x == 'inf' else float, input("enter third row ").split(" ")))
    fourth_row = list(map(lambda x: infinity if x == 'inf' else float, input("enter the fourth row ").split(" ")))
    adjacency_matrix = [first_row,
             second_row,
             third_row,
             fourth_row]
    # get the distance matrix by passing the graph as input to the floyd warshall function
    distance_matrix = floyd_warshall(adjacency_matrix)
    # print the result
    for i in distance_matrix:
        for j in i:
            if j != float('inf'):
                print(j, '\t', end='')
            else:
                print(j, end=' ')
        print()

File: Graphs/FloydWarshall.scala

/*Implementation of Floyd Warshall Algorithm in Scala*/

object Main {
    //Initialization of variables
    var INF: Int = 99999
    var V: Int = 4
    
    //Function for all pair shortest path
    def FloydWarshall(graph: Array[Array[Int]]): Unit =
    {
        /*dist array stores the final result which stores
         the shortest path between every pair of vertices*/
        var dist = Array.ofDim[Int](V,V)
        //Initialization of local variables
        var i: Int = 0
        var j: Int = 0
        var k: Int = 0
        
        //Initialized the resultant matrix to the original matrix
        for(i <- 0 until V)
        {
            for(j <- 0 until V)
            {
                dist(i)(j) = graph(i)(j)
            }
        }
        //Adding all the vertices one by one
        for(k <- 0 until V)
        {
            for(i <- 0 until V)
            {
                for(j <- 0 until V)
                {
                    /*If the length between the vertices is
                    shorter than update the distance matrix*/
                    if(dist(i)(k) + dist(k)(j) < dist(i)(j))
                    {
                        dist(i)(j) = dist(i)(k) + dist(k)(j)
                    }
                }
            }
        }
        
        //Displaying the resultant matrix 
        for(i <- 0 until V)
        {
            for(j <- 0 until V)
            {
                if(dist(i)(j) == INF)
                {
                    println("INF ")
                }
                else
                {
                    println(dist(i)(j)+ "  ")
                }
            }
        }
    }
    
    //Driver code
	def main(args: Array[String]): Unit =
	{
	    //Giving the input matrix
		var graph = Array( Array(0,   5,  INF, 10), 
                            Array(INF, 0,   3, INF), 
                            Array(INF, INF, 0,   1), 
                            Array(INF, INF, INF, 0) 
                          );
        //Function calling
        FloydWarshall(graph)
	}
}